### ISSUE TITLE
feat: declare mocked objects as object properties directly

### ISSUE THREAD

**Issue by nopenopenope** [paul-thebaud/phpunitgen-core#20]
### Subject of the issue

Currently, mocked objects properties are declared like so:
```
    /**
     * @var Context|Mock
     */
    protected $context;
```

Since Mockery and PhpUnit are able to read from the explicit or implicit interface, we don't need PhpDocs here. Instead, the property declaration should be like this:
```
    protected object $context;
```
This allows Coding Standard tools like PhpStan or ECS to check the property, but at the same time spare 3 lines of PhpDocs that blow up somewhat more extended tests. The outcome will be a smaller code line count whilst keeping the Type-hint functionalities of tools like PhpStorm. This will increase the quality of the generated tests automatically IMO.

The only exception should be made to the to be unit-tested property, which should be declared with its Type or if possible, the interface. This way, the Type Hinting in IDEs will work neatly as expected and better code can be generated. 

Let me hear your thoughts. :) 

tl;dr: I suggest to declare properties as `object` and remove the PhpDoc, as Type Hinting will work automatically by tools provided by Mockery/PhpUnit.

### Your environment
* version of `phpunitgen/core`: 1
* version of `phpunitgen/console`: 1.5

**Comment by paul-thebaud** [comment]
Hi,
This behavior is more complicated, because PHP 7 and 8 features are different on typed properties capabilities (eg. typed properties from 7.4, union types from 8.0, intersection types from 8.1).
I'll try what phpunit and mockery support on this and try to think of a good way to support it too.
In all cases, it may take some time to release a such feature.
Thanks for posting your thoughts 🙂

**Comment by paul-thebaud** [comment]
Hi @nopenopenope,
Just to keep in touch, I'm currently on this feature, and it's based on a new configuration option which makes the test class properties typed when enabled (it is disabled by default).
